src/java.base/share/classes/java/lang/invoke/MethodHandleStatics.java

```diff
@@ -58,6 +62,11 @@ private MethodHandleStatics() { }  // do not instantiate
     static final int MAX_ARITY;
     static final boolean VAR_HANDLE_IDENTITY_ADAPT;
 
+    // Markers used by LambdaForm and Species resolution code to communicate
+    // classes that have been or could have be pre-generated
+    static final String LF_RESOLVE = "[LF_RESOLVE]";
+    static final String SPECIES_RESOLVE = "[SPECIES_RESOLVE]";
```

Comment:
It is okay with me to simply hardcode these strings in GenerateJLIClassesHelper and the trace methods.